core/include/esp/sar_regs.h

```diff
@@ -0,0 +1,39 @@
+/* esp/sar_regs.h
+ *
+ * ESP8266 register definitions for the "sar" region (0x3FF2xxx)
+ *
+ * The 0x60000D00 register region is referred to as "sar" by some old header
+ * files. Apparently referenced both by ROM I2C functions as well as ADC
+ * config/read functions.
+ *
+ * Not compatible with ESP SDK register access code.
+ */
+
+#ifndef _ESP_SAR_REGS_H
+#define _ESP_SAR_REGS_H
+
+#include "esp/types.h"
+#include "common_macros.h"
+
+#define SAR_BASE 0x60000d00
+// Unfortunately,
+// esp-open-sdk/xtensa-lx106-elf/xtensa-lx106-elf/sysroot/usr/include/xtensa/config/specreg.h
+// already has a "SAR" macro definition which would conflict with this, so
+// we'll use "ESPSAR" instead..
+#define ESPSAR (*(struct SAR_REGS *)(SAR_BASE))
+
+/* Note: This memory region is not currently well understood.  Pretty much all
+ * of the definitions here are from reverse-engineering the Espressif SDK code,
+ * many are just educated guesses, and almost certainly some are misleading or
+ * wrong.  If you can improve on any of this, please contribute!
+ */
+
+struct SAR_REGS {
+    uint32_t volatile _unknown0[18];   // 0x00 - 0x44
+    uint32_t volatile UNKNOWN_48;      // 0x48 : used by sdk_system_restart_in_nmi()
```

Comment:
These are not inconsistant.  "_unknown" is explicitly intended to be a temporary placeholder for spacing purposes and should not be referenced by any code (there may be one or two places that do reference them, but if there is that's an error and needs to be cleaned up).

The `UNKNOWN_48` is an example of the (rare) situation where a field needed to be referenced somewhere but I still could not figure out what it should be named from the context.  The uppercase indicates it's not just a spacer/placeholder but is actually a referenced field.

If you want to go through and clean up the "_unknown" names with underscores/leading digits/etc, you certainly can, but frankly it seems pretty minor to me, considering that they are explicitly not supposed to be being referenced by anything (but if somebody wants to do a PR for that that's fine by me)..